### PR TITLE
add person sensor (sen21231) from useful sensors 

### LINKED ISSUE #1925: Add support for the Person Sensor by Useful Sensors (SEN-21231)
**Describe the problem you have/What new integration you would like**
Please add support for the Person Sensor by Useful Sensors (SEN-21231) so that we can have good presence detention using ML sensor

**Please describe your use case for this integration and alternatives you've tried:**
 i've been using IR and Microwave sensor for presence detection but it is not that reliable. using this sensor with built in ML can elevate presence sensor to the next level. And the cost of the sensor is relatively affordable, USD 10 

**Additional context**
 module info : https://github.com/usefulsensors/person_sensor_docs

### PR DESCRIPTION
**Pull request by shreyaskarnik**
## Description:

**Related issue (if applicable):** fixes https://github.com/esphome/feature-requests/issues/1925

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#<esphome PR number goes here>

## Checklist:

  - [x] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [x] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

**Comment by jesserockz** [comment]
As this is a feature matched with a PR in `esphome/esphome`, please target your PR to the `next` branch and rebase.

Thanks
Jesse

**Comment by shreyaskarnik** [comment]
Thanks, for the feedback @jesserockz I updated the target branch and rebased but could not fix the change in `_static/version`. Happy to add a new PR for the docs. Apologies about targeting the current branch did not know about the next branch.

File: components/sensor/sen21231.rst [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+
+    Person Sensor
+
+This sensor returns information over I2C interface. The I2C address is ``0x62``
+
+.. code-block:: yaml
+
+    # Example configuration entry
+    output:
```
Comment (jesserockz) [inline components/sensor/sen21231.rst]: This is a sensor, not an output...
```suggestion
    sensor:
```

File: components/sensor/sen21231.rst [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      - platform: sen21231
+        id: person_sensor
+        # sda and scl connected approriately
+        # power connected to 3.3v
```
Comment (jesserockz) [inline components/sensor/sen21231.rst]: We don't need extra comments here when this information is contained in the i2c docs.
```suggestion
```

File: components/sensor/sen21231.rst [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+`Person Sensor (SEN21231) from Useful Sensors <https://usefulsensors.com/person-sensor>`__ has built in facial recognition that can detect how many people are facing the sensor and relative positions of the faces.
+
+.. figure:: images/sen21231.png
+    :align: center
+    :width: 70.0%
+
+
+    Person Sensor
+
+This sensor returns information over I2C interface. The I2C address is ``0x62``
```
Comment (jesserockz) [inline components/sensor/sen21231.rst]:
```suggestion
This sensor requires an :doc:`/components/i2c`. The I2C address is ``0x62``
```

**Review by jesserockz — CHANGES_REQUESTED** [review]

**Comment by shreyaskarnik** [comment]
@jesserockz thanks for the suggestions I have applied them. I appreciate your help. 🙌

**Review by jesserockz — APPROVED** [review]

**Pull request merged**